File: Python/Binary Trees/L_evaluateTree.py
def evaluateTree(root) -> bool:
        if root.val == 0:
            return False
        elif root.val == 1:
            return True
        elif root.val == 2:
            return evaluateTree(root.left) or evaluateTree(root.right)
        elif root.val == 3:
            return evaluateTree(root.left) and evaluateTree(root.right)

File: cpp/Binary Tree/L_evaluateTree.cpp
#include<bits/stdc++.h>
using namespace std;

bool evaluateTree(TreeNode* root) {
    if(root->val == 0) return false;
    if(root->val == 1) return true;
    if(root->val == 2) return evaluateTree(root->left) || evaluateTree(root->right);
    if(root->val == 3) return evaluateTree(root->left) && evaluateTree(root->right);
    return NULL;
}

int main() {

}